administration/config/env.ts

```diff
@@ -64,10 +66,10 @@ function getClientEnvironment(publicUrl: string): {
   const raw = Object.keys(process.env)
     .filter(key => REACT_APP.test(key))
     .reduce(
-      (env: Record<string, string | boolean | undefined>, key: keyof typeof process.env) => {
-        env[key] = process.env[key]
-        return env
-      },
+      (env: Record<string, string | boolean | undefined>, key: keyof typeof process.env) => ({
+        ...env,
+        [key]: process.env[key],
+      }),
```

Comment:
Can you explain why this should be reverted? In general it is not good practice to reassign params/objects, I think code has less side effects if you only work with new copies.
The linting rule is this one:
https://eslint.org/docs/latest/rules/no-param-reassign

Reply:
It creates many unnecessary copies of the object and is just unnecessarily slower (likely quadratic-ish runtime instead of linear-ish) and generates higher load for the garbage collector.

Also, in this kind of reduce functions, it is quite clear that mutations are totally fine.

I think I am, in general, not a huge fan of this linting rule :D

Reply:
Also, the motivation of the rule only references the [function arguments object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) which apparently does not even exist for arrow functions.

Reply:
Works for me, don't really mind. I personally like the destructuring syntax more and think it's less error prone, but that is definitely a matter of personal preferences. I also see your point in terms of runtime and garbage collection, not sure if that has any noticeable effect in our case though.